src/js/objects/ChartComponents.js

```diff
@@ -288,7 +288,7 @@ let componentConfigs = {
 					data.xPositions[j],
 					y,
 					data.barWidth,
-					c.color,
+					data.colors ? ((j < data.colors.length) ? data.colors[j] : data.colors[0]) : Array.isArray(c.color) ? (j < c.color.length ? c.color[j] : c.color[0]) : c.color,
```

Comment:
Could you add a comment telling what's happening here. Nested ternary statements is a lot to process.